libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/location/NavigationLocationProvider.kt

```diff
@@ -53,7 +57,10 @@ import java.util.concurrent.CopyOnWriteArraySet
  * }
  * ```
  */
-class NavigationLocationProvider : LocationProvider {
+open class NavigationLocationProvider :
+    LocationProvider,
+    LocationObserver,
```

Comment:
By implementing the interface directly, we're exposing it as part of the `NavigationLocationProvider` surface, while it's not needed. The `LocationObserver` instance could be a member of the `NavigationLocationProvider` instead.